os/arch/arm/src/amebasmart/amebasmart_i2c.c

```diff
@@ -923,7 +925,16 @@ static int amebasmart_i2c_init(FAR struct amebasmart_i2c_priv_s *priv)
 	/* Power-up and configure GPIOs */
 	DEBUGASSERT(priv);
 	DEBUGASSERT(!priv->i2c_object);
-	priv->i2c_object = (i2c_t *)kmm_malloc(sizeof(i2c_t));
+
+	if (!priv->initialized) {
+		priv->i2c_object = (i2c_t *)kmm_malloc(sizeof(i2c_t));
+		DEBUGASSERT(priv->i2c_object);
+		priv->initialized = true;
+	} else {
+		/* memset 0 if not required to allocate */
+		memset(priv->i2c_object, 0, sizeof(i2c_t));
```

Comment:
The `i2c_t` has just two members with 32 bit size of each. Then, just allocating 0 or null will be much faster than memset which is working with 8 bit. How do you think?